backend/endpoints/rooms.py

```diff
@@ -61,3 +63,69 @@ async def create_room(
         status_code=status.HTTP_424_FAILED_DEPENDENCY,
         detail="unable to create room",
     )
+
+
+@router.put(
+    "/org/{org_id}/rooms/{room_id}/members/{member_id}",
+    status_code=status.HTTP_200_OK,
+    responses={
+        400: {"detail": "the max number for a Group_DM is 9"},
+        401: {"detail": "member not an admin"},
+        403: {"detail": "DM room or not found"},
+    },
+)
+async def add_to_room(data: AddToRoom, org_id: str, room_id: str, member_id: str):
+    """Adds a new member(s) to a room
+    Args:
+        data: a pydantic schema that defines the request params
+        org_id (str): A unique identifier of an organisation
+        room_id: A unique identifier of the room to be updated
+
+    Returns:
+        HTTP_200_OK: member added
+    Raises:
+        HTTP_400_BAD_REQUEST: the max number for a Group_DM is 9
+        HTTP_401_UNAUTHORIZED: member not an admin
+        HTTP_403_FORBIDDEN: DM room or not found
+    """
+
+    new_member = data.dict().get("new_member")
+    room = await get_room(org_id=org_id, room_id=room_id)
+
+    if room is not None and room["room_type"] != RoomType.DM:
+        member = room.get("room_members").get(str(member_id))
+
+        if member["role"] == Role.ADMIN:
+            room_members = room["room_members"]
+
+            if room["room_type"] == RoomType.CHANNEL:
+                room_members.update(new_member)
+
+            if room["room_type"] == RoomType.GROUP_DM:
+                for person in list(new_member.keys()):
+                    if len(room_members.keys()) >= 9:
+                        raise HTTPException(
+                            detail="the max number for a Group_DM is 9",
+                            status_code=400,
+                        )
+                    new_data = {person: new_member.get(str(person))}
+                    room_members.update(new_data)
+
+            update_members = {"room_members": room_members}
+            update_res = await DB.update(
+                "chat_rooms", document_id=room_id, data=update_members
+            )
+            if update_res and update_res.get("status_code", None) is None:
```

Comment:
Centrifugo functions can be put in a background task so that they do not affect the endpoint response.
Because centrifugo maybe down, but in essence the data should be saved in the core.
Therefore, centrifugo being up or down shouldn't affect the response, as long as the data is saved to core, it should return a success response, and the call to centrifugo shouldn't block the request

Reply:
I agree. I will see to that too. Thanks